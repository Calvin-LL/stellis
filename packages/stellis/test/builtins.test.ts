import { describe, it, expect } from 'vitest';
import transformStellis from './utils';

describe('error-boundary', () => {
  it('should transform', async () => {
    const code = `
    <stellis:error-boundary
      fallback={(error) => <>
        <h1>Error: {error.name}</h1>
        <p>Message: {error.message}</p>
      </>}
    >
      <FailingComponent />
    </stellis:error-boundary>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
describe('fragment', () => {
  it('should transform', async () => {
    const code = `
    <stellis:fragment set:html="<script>Hello World</script>" />
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
describe('comment', () => {
  it('should transform', async () => {
    const code = `
      <stellis:comment value="This is a comment." />
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
describe('head', () => {
  it('should transform', async () => {
    const code = `
      <stellis:head type="pre">
        <title>Hello World</title>
      </stellis:head>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
describe('body', () => {
  it('should transform', async () => {
    const code = `
      <stellis:body type="post">
        <script src="./my-script.js" />
      </stellis:body>
    `;

    expect(await transformStellis(code)).toMatchSnapshot();
  });
});
