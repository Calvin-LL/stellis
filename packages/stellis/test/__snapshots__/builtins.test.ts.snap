// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`body > should transform 1`] = `
"import { $$html as _$$html } from \\"stellis\\";
import { $$inject as _$$inject } from \\"stellis\\";
const _template = [\\"<script src=\\\\\\"./my-script.js\\\\\\">\\", \\"</script>\\"];
_$$inject(\\"body\\", {
  type: \\"post\\",
  children: _$$html(_template, true, [])
});"
`;

exports[`comment > should transform 1`] = `
"import { $$comment as _$$comment } from \\"stellis\\";
_$$comment({
  value: \\"This is a comment.\\"
});"
`;

exports[`error-boundary > should transform 1`] = `
"import { $$component as _$$component } from \\"stellis\\";
import { $$html as _$$html } from \\"stellis\\";
import { $$errorBoundary as _$$errorBoundary } from \\"stellis\\";
const _template = [\\"<h1>\\", \\"</h1>\\"],
  _template2 = [\\"<p>\\", \\"</p>\\"];
_$$errorBoundary({
  fallback: error => [_$$html(_template, true, () => [\\"Error: \\", error.name]), _$$html(_template2, true, () => [\\"Message: \\", error.message])],
  children: _$$component(FailingComponent, {})
});"
`;

exports[`fragment > should transform 1`] = `
"import { $$fragment as _$$fragment } from \\"stellis\\";
_$$fragment({
  \\"set:html\\": \\"<script>Hello World</script>\\"
});"
`;

exports[`head > should transform 1`] = `
"import { $$html as _$$html } from \\"stellis\\";
import { $$inject as _$$inject } from \\"stellis\\";
const _template = \\"<title>Hello World</title>\\";
_$$inject(\\"head\\", {
  type: \\"pre\\",
  children: _$$html(_template, true)
});"
`;
